### PR TITLE
Use authopen binary to get read/write access to USB disks

### PR DESCRIPTION
**Pull request by grulja**
This makes it work on Mac OSX Catalina

Initial code made by @MartinBriza

Closes https://github.com/FedoraQt/MediaWriter/issues/217

### CODE REVIEW

File: helper/mac/writejob.cpp [hunk 1]
```diff
@@ -80,24 +155,21 @@ void WriteJob::onFileChanged(const QString &path) {
...
 
     QProcess diskUtil;
     diskUtil.setProgram("diskutil");
-    diskUtil.setArguments(QStringList() << "unmountDisk" << where);
-    diskUtil.start();
-    diskUtil.waitForFinished();
 
     source.open(QIODevice::ReadOnly);
-    target.open(QIODevice::WriteOnly);
+    target.open(fd, QIODevice::WriteOnly, QFileDevice::AutoCloseHandle);
```
Comment (MartinBriza) [inline helper/mac/writejob.cpp:172]: You have `AutoCloseHandle` here... (continued below)

File: helper/mac/writejob.cpp [hunk 2]
```diff
@@ -210,9 +282,9 @@ bool WriteJob::writeCompressed() {
     }
 }
 
-void WriteJob::check() {
-    QFile target("/dev/r"+where);
-    target.open(QIODevice::ReadOnly);
+void WriteJob::check(int fd) {
+    QFile target;
+    target.open(fd, QIODevice::ReadOnly, QFileDevice::AutoCloseHandle);
```
Comment (MartinBriza) [inline helper/mac/writejob.cpp:287]: But you're using `AutoCloseHandle` here as well, after the previous `QFile` has closed the FD. I think it should be only in this function but not in the write ones.

**Comment by meelash** [comment]
I'm not sure where to report this, so I'm posting here. The release available on getfedora.org is still the 4.1.4 version which has this bug. Anyway to get them to bump it to 4.1.6, sure would save a lot of pain. I spent an hour debugging UEFI problems before realizing the dang USB had never been written!

**Comment by dreua** [comment]
@meelash I think this would be the right place: https://pagure.io/fedora-websites/issues
Unfortunately I can't log in from mobile right now.

**Comment by meelash** [comment]
Done.
https://pagure.io/fedora-web/websites/issue/136

**Pull request merged**